waddrmgr/manager_test.go

```diff
@@ -1019,38 +1019,64 @@ func testImportScript(tc *testContext) bool {
 
 // testMarkUsed ensures used addresses are flagged as such.
 func testMarkUsed(tc *testContext) bool {
-	expectedAddr1 := expectedAddr{
-		addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
-		used:        true,
+	tests := []struct {
+		name       string
+		addrType   btcutil.Address
```

Comment:
I'm using this field now, to switch between `managedAddress` and `scriptAddress` so that both paths are covered in the test cases.